src/libraries/Microsoft.Extensions.Caching.Abstractions/src/CacheEntryExtensions.cs

```diff
@@ -172,12 +181,24 @@ public static ICacheEntry SetOptions(this ICacheEntry entry, MemoryCacheEntryOpt
                 entry.AddExpirationToken(expirationToken);
             }
 
-            foreach (PostEvictionCallbackRegistration postEvictionCallback in options.PostEvictionCallbacks)
+            for (int i = 0; i < options.PostEvictionCallbacks.Count; i++)
             {
-                entry.RegisterPostEvictionCallback(postEvictionCallback.EvictionCallback!, postEvictionCallback.State);
+                PostEvictionCallbackRegistration postEvictionCallback = options.PostEvictionCallbacks[i];
+                if (postEvictionCallback.EvictionCallback is null)
+                    ThrowNullCallback(i, nameof(options));
+
+                entry.RegisterPostEvictionCallbackNoValidation(postEvictionCallback.EvictionCallback, postEvictionCallback.State);
             }
 
             return entry;
         }
+
+        [DoesNotReturn]
+        private static void ThrowNullCallback(int index, string paramName)
+        {
+            string message =
+                $"MemoryCacheEntryOptions.PostEvictionCallbacks contains a PostEvictionCallbackRegistration with a null EvictionCallback at index {index}.";
```

Comment:
This should be in a RESX, but we don't do that for other exception messages in this library, so I guess we can just follow the same pattern here.

Reply:
There is no RESX for this package, all exceptions are written with interpolation